tests/components/sonos/test_sensor.py

```diff
@@ -75,7 +75,7 @@ async def test_battery_on_S1(hass, config_entry, config, soco, battery_event):
 
     # Update the speaker with a callback event
     speaker = hass.data[DATA_SONOS].discovered[soco.uid]
-    speaker.async_dispatch_properties(battery_event)
+    speaker.async_dispatch_event(battery_event)
```

Comment:
It would be good to rewrite the tests as done in the switch platform, to get the callback from the library mock. Then we wouldn't need to change the tests when refactoring the integration.

Reply:
Good call. Set up proper/explicit mocks for all currently used Sonos `Service` instances to make this easier moving forward.